src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -234,17 +279,50 @@ private uint GetPlaceholderScope(BoundValuePlaceholderBase placeholder)
         public override BoundNode? Visit(BoundNode? node)
         {
 #if DEBUG
-            if (node is BoundValuePlaceholderBase placeholder
-                // CheckValEscapeOfObjectInitializer() does not use BoundObjectOrCollectionValuePlaceholder.
-                // CheckInterpolatedStringHandlerConversionEscape() does not use BoundInterpolatedStringHandlerPlaceholder.
-                && node is not (BoundObjectOrCollectionValuePlaceholder or BoundInterpolatedStringHandlerPlaceholder))
+            if (node is BoundExpression expr)
+            {
+                bool added = MarkVisited(expr);
```

Comment:
nit: there's something slightly odd about marking as visited before we've completed the visiting, but I understand this helps for checks in `Get/CheckEscape` methods...